### PR TITLE
Upgrades to base image.

### PR DESCRIPTION
**Pull request by tgxworld**

### CODE REVIEW

File: image/base/install-imagemagick [hunk 1]
```diff
@@ -17,7 +17,7 @@ cd $WDIR/libpng
 make all && make install
 
 # Build and install ImageMagick
-wget -O $WDIR/ImageMagick.tar.gz "http://www.imagemagick.org/download/ImageMagick.tar.gz"
+wget -O $WDIR/ImageMagick.tar.gz "http://www.imagemagick.org/download/ImageMagick-6.9.4-1.tar.gz"
```
Comment (tgxworld) [inline image/base/install-imagemagick]: ImageMagick-7 has been released and rebuilding a new image resulted in ImageMagick being upgraded to 7.

File: image/base/install-pngcrush [hunk 2]
```diff
@@ -2,7 +2,7 @@
 set -e
 PNGCRUSH_VERSION=1.7.92
 cd /tmp
-curl -O http://iweb.dl.sourceforge.net/project/pmt/pngcrush/$PNGCRUSH_VERSION/pngcrush-$PNGCRUSH_VERSION.tar.gz
+wget http://iweb.dl.sourceforge.net/project/pmt/pngcrush/$PNGCRUSH_VERSION/pngcrush-$PNGCRUSH_VERSION.tar.gz
```
Comment (tgxworld) [inline image/base/install-pngcrush]: Getting redirected here. I could pass `-L` to curl but all the other scripts are using `wget` so I'm making it consistent

**Comment by tgxworld** [comment]
@SamSaffron Please review :) I've tested it out on two of my own instance and got it upgraded and running :+1:

**Comment by tgxworld** [comment]
@SamSaffron Updated the base image to use ruby 2.3.1.

File: image/base/Dockerfile [hunk 3]
```diff
@@ -81,10 +81,10 @@ RUN add-apt-repository ppa:ubuntu-toolchain-r/test &&\
     echo 'gem: --no-document' >> /usr/local/etc/gemrc &&\
     mkdir /src && cd /src && git clone https://github.com/sstephenson/ruby-build.git &&\
     cd /src/ruby-build && ./install.sh &&\
-    cd / && rm -rf /src/ruby-build && ruby-build 2.0.0-p647 /usr/local &&\
+    cd / && rm -rf /src/ruby-build && ruby-build 2.3.1 /usr/local &&\
     gem update --system &&\
     gem install bundler &&\
-    rm -rf /usr/local/share/ri/2.0.0/system &&\
```
Comment (tgxworld) [inline image/base/Dockerfile]: `lsb-releases` does not come installed with the base image.

`sudo` is required before https://github.com/tgxworld/discourse_docker/blob/ddf77f732f8a9fbd5c048c531a5fbc942e7438b5/image/base/Dockerfile#L27

**Comment by fantasticfears** [comment]
Cloudflare releases a patch to enable http2 modules and spdy. It builds against 1.9.7. The reason they built this is many old browsers can't understand h2. Don't know whether this would be useful. Hope it may helps.

https://blog.cloudflare.com/open-sourcing-our-nginx-http-2-spdy-code/

**Pull request merged**